sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -85,6 +88,26 @@ public bool IsRunning
                     return _isRunningOverride.Value;
                 }
 
+                var status = Status;
+                return ((status == EventProcessorStatus.Running) || (status == EventProcessorStatus.Stopping));
+            }
+
+            protected set => _isRunningOverride = value;
+        }
+
+        /// <summary>
+        ///   Indicates the current state of the processor.
+        /// </summary>
+        ///
+        internal EventProcessorStatus Status
+        {
+            get
+            {
+                if (_statusOverride.HasValue)
```

Comment:
Good catch; that's definitely a race.   Looking at it with fresh eyes, it also could get out of sync with the start/stop flow.   Tweaked things a bit to read that within the semaphore, if acquired, which should give us additional confidence since we're only now setting it when the semaphore is held.

Good idea?  No so good idea?

Reply:
I think it is an improvement!